core/src/test/java/org/pentaho/platform/dataaccess/datasource/api/DataSourceWizardServiceTest.java

```diff
@@ -264,43 +268,34 @@ public void testRemoveDSWError() throws Exception {
   }
 
   @Test
-  public void testGetDSWDatasourceIds() throws Exception {
-    final String TEST_DOMAIN_ID = "mock";
-
-    LogicalModelSummary mockLogicalModelSummary  = mock( LogicalModelSummary.class );
-    doReturn(TEST_DOMAIN_ID).when(mockLogicalModelSummary).getDomainId();
-    List<LogicalModelSummary> mockLogicalModelSummaryList = new ArrayList<LogicalModelSummary>();
-    mockLogicalModelSummaryList.add( mockLogicalModelSummary );
-
-    List<String> datasourceList = new ArrayList<String>();
-    datasourceList.add( mockLogicalModelSummary.getDomainId() );
-
-    doReturn( mockLogicalModelSummaryList ).when( dataSourceWizardService.dswService ).getLogicalModels( null );
-
-    Map<String,InputStream> mockDomainFilesData = new HashMap<String,InputStream>();
-
-    mockDomainFilesData.put(TEST_DOMAIN_ID, IOUtils.toInputStream("AGILE_BI_GENERATED_SCHEMA"));
-    doReturn(mockDomainFilesData).when((IPentahoMetadataDomainRepositoryExporter)dataSourceWizardService.metadataDomainRepository).getDomainFilesData(anyString());
-
-    List<String> datasourceIds = dataSourceWizardService.getDSWDatasourceIds();
-    verify( dataSourceWizardService, times( 1 ) ).getDSWDatasourceIds();
-    assert( datasourceIds.contains(TEST_DOMAIN_ID));
+  public void testGetDSWDatasourceIds() {
+    List<String> mockMetadataIdsList = new ArrayList<String>();
+    Set<String> mockSet = new HashSet<String>();
+    mockSet.add( "domainId1" );
+    mockMetadataIdsList.add( "domainId1" );
+    doReturn( 1 ).when( dataSourceWizardService ).noOfThreads();
+    doReturn( "domainId1" ).when( dataSourceWizardService ).isDataSourceWizard( "domainId1" );
+    doReturn( mockSet ).when( dataSourceWizardService.metadataDomainRepository ).getDomainIds();
+
+    List<String> response = dataSourceWizardService.getDSWDatasourceIds();
+    assertEquals( mockMetadataIdsList, response );
   }
 
   @Test
   public void testGetDSWDatasourceIdsError() throws Exception {
-    doReturn( null ).when( dataSourceWizardService.dswService ).getLogicalModels( null );
+    doReturn( null ).when( dataSourceWizardService.metadataDomainRepository ).getDomainIds();
```

Comment:
this method would never return null, it would return an empty set
https://github.com/pentaho/pentaho-platform/blob/master/extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/PentahoMetadataDomainRepository.java#L509-L527

PentahoMetadataDomainRepository#getDomainIds